pkg/api/kubegraph/nodes/types.go

```diff
@@ -267,3 +270,56 @@ func (*HorizontalPodAutoscalerNode) Kind() string {
 func (n HorizontalPodAutoscalerNode) UniqueName() osgraph.UniqueName {
 	return HorizontalPodAutoscalerNodeName(n.HorizontalPodAutoscaler)
 }
+
+func PetSetNodeName(o *kapps.PetSet) osgraph.UniqueName {
+	return osgraph.GetUniqueRuntimeObjectNodeName(PetSetNodeKind, o)
+}
+
+type PetSetNode struct {
+	osgraph.Node
+	*kapps.PetSet
+}
+
+func (n PetSetNode) Object() interface{} {
+	return n.PetSet
+}
+
+func (n PetSetNode) String() string {
+	return string(n.UniqueName())
+}
+
+func (n PetSetNode) UniqueName() osgraph.UniqueName {
+	return PetSetNodeName(n.PetSet)
+}
+
+func (*PetSetNode) Kind() string {
+	return PetSetNodeKind
+}
+
+func PetSetSpecNodeName(o *kapps.PetSetSpec, ownerName osgraph.UniqueName) osgraph.UniqueName {
+	return osgraph.UniqueName(fmt.Sprintf("%s|%v", PetSetSpecNodeKind, ownerName))
+}
+
+type PetSetSpecNode struct {
+	osgraph.Node
+	*kapps.PetSetSpec
```

Comment:
Just following bad patterns set by the person before me.  Will fix most of
them.

On Thu, Aug 4, 2016 at 2:42 PM, David Eads <notifications@github.com> wrote:

> In pkg/api/kubegraph/nodes/types.go
> <https://github.com/openshift/origin/pull/9972#discussion_r73579653>:
>
> > +
> > +func (n PetSetNode) UniqueName() osgraph.UniqueName {
> > +	return PetSetNodeName(n.PetSet)
> > +}
> > +
> > +func (*PetSetNode) Kind() string {
> > +	return PetSetNodeKind
> > +}
> > +
> > +func PetSetSpecNodeName(o *kapps.PetSetSpec, ownerName osgraph.UniqueName) osgraph.UniqueName {
> > +	return osgraph.UniqueName(fmt.Sprintf("%s|%v", PetSetSpecNodeKind, ownerName))
> > +}
> > +
> > +type PetSetSpecNode struct {
> > +	osgraph.Node
> > +	*kapps.PetSetSpec
>
> non-anonymous please.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9972/files/fdbb46f05fe687c7d2791edf8f685f181e355e91#r73579653>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_pxR2YUC979ZEYGjeH2KqseVOPMTmks5qcjKJgaJpZM4JRYA_>
> .
>